pkg/operator/sync.go

```diff
@@ -171,6 +172,36 @@ func (optr *Operator) syncMachineConfigController(config renderConfig) error {
 	return nil
 }
 
+func (optr *Operator) syncEtcdQuorumGuard(config renderConfig) error {
+	eqgBytes, err := renderAsset(config, "manifests/etcdquorumguard/deployment.yaml")
+	if err != nil {
+		return err
+	}
+	eqg := resourceread.ReadDeploymentV1OrDie(eqgBytes)
+
+	_, updated, err := resourceapply.ApplyDeployment(optr.kubeClient.AppsV1(), eqg)
+	if err != nil {
+		return err
+	}
+	if updated {
+		var waitErrs []error
+		waitErrs = append(waitErrs, optr.waitForDeploymentRollout(eqg))
+		if len(waitErrs) > 0 {
+			agg := utilerrors.NewAggregate(waitErrs)
+			return agg
+		}
```

Comment:
all of this inside `if updated {` should just just be:

```
if updated {
    if err := optr.waitForDeploymentRollout(eqg); err != nil {
        return err
    }
}
```
Otherwise you'll end appending `nil` to an empty slice, which is going to result in `len(waitErrs) > 0` as you can see here https://play.golang.org/p/Bm9oGH45YsP